src/core/execute.c

```diff
@@ -2735,6 +2752,28 @@ static void invalid_env(const char *p, void *userdata) {
         log_unit_error(info->unit, "Ignoring invalid environment assignment '%s': %s", p, info->path);
 }
 
+int exec_context_named_iofds(Unit *unit, const ExecContext *c, const ExecParameters *p, int *named_iofds) {
+        unsigned int i, targets = 0;
+        targets += (c->std_input == EXEC_INPUT_NAMED_FD) ? 1 : 0;
+        targets += (c->std_output == EXEC_OUTPUT_NAMED_FD) ? 1 : 0;
+        targets += (c->std_error == EXEC_OUTPUT_NAMED_FD) ? 1 : 0;
```

Comment:
Yeah, we do it all the time, and I think reducing the clutter (repeated assignment, ternary op…) makes it easier to see the important part.